File: Clients/Node/Core/src/models/videos.ts
/**
 * Video generation models and types for the Conduit Core API
 */

/**
 * Request for generating a video from a text prompt
 */
export interface VideoGenerationRequest {
  /** The text prompt that describes what video to generate */
  prompt: string;
  
  /** The model to use for video generation (e.g., "minimax-video") */
  model?: string;
  
  /** The duration of the video in seconds. Defaults to 5 seconds */
  duration?: number;
  
  /** The size/resolution of the video (e.g., "1920x1080", "1280x720") */
  size?: string;
  
  /** Frames per second for the video. Common values: 24, 30, 60 */
  fps?: number;
  
  /** The style or aesthetic of the video generation */
  style?: string;
  
  /** The format in which the generated video is returned. Options: "url" (default) or "b64_json" */
  response_format?: 'url' | 'b64_json';
  
  /** A unique identifier representing your end-user */
  user?: string;
  
  /** Optional seed for deterministic generation */
  seed?: number;
  
  /** The number of videos to generate. Defaults to 1 */
  n?: number;
}

/**
 * Response from a video generation request
 */
export interface VideoGenerationResponse {
  /** Unix timestamp of when the response was created */
  created: number;
  
  /** List of generated video data */
  data: VideoData[];
  
  /** The model used for generation */
  model?: string;
  
  /** Usage information if available */
  usage?: VideoUsage;
}

/**
 * A single generated video
 */
export interface VideoData {
  /** The URL of the generated video, if response_format is "url" */
  url?: string;
  
  /** The base64-encoded video data, if response_format is "b64_json" */
  b64_json?: string;
  
  /** The revised prompt that was used for generation */
  revised_prompt?: string;
  
  /** Additional metadata about the generated video */
  metadata?: VideoMetadata;
}

/**
 * Usage statistics for video generation
 */
export interface VideoUsage {
  /** The number of prompt tokens used */
  prompt_tokens: number;
  
  /** The total number of tokens used */
  total_tokens: number;
  
  /** The duration processed in seconds */
  duration_seconds?: number;
  
  /** The total processing time in seconds */
  processing_time_seconds?: number;
}

/**
 * Metadata about a generated video
 */
export interface VideoMetadata {
  /** The actual duration of the generated video in seconds */
  duration?: number;
  
  /** The resolution of the generated video */
  resolution?: string;
  
  /** The frames per second of the generated video */
  fps?: number;
  
  /** The file size in bytes */
  file_size_bytes?: number;
  
  /** The video format/codec */
  format?: string;
  
  /** The seed used for generation, if any */
  seed?: number;
}

/**
 * Request for async video generation
 */
export interface AsyncVideoGenerationRequest extends VideoGenerationRequest {
  /** The webhook URL to receive the result when generation is complete */
  webhook_url?: string;
  
  /** Additional metadata to include with the webhook callback */
  webhook_metadata?: Record<string, any>;
  
  /** The timeout for the generation task in seconds */
  timeout_seconds?: number;
}

/**
 * Response from an async video generation request
 */
export interface AsyncVideoGenerationResponse {
  /** The unique task identifier */
  task_id: string;
  
  /** The current status of the task */
  status: VideoTaskStatus;
  
  /** The progress percentage (0-100) */
  progress: number;
  
  /** An optional progress message */
  message?: string;
  
  /** The estimated time to completion in seconds */
  estimated_time_to_completion?: number;
  
  /** When the task was created */
  created_at: string;
  
  /** When the task was last updated */
  updated_at: string;
  
  /** The generation result, available when status is Completed */
  result?: VideoGenerationResponse;
  
  /** Error information if the task failed */
  error?: string;
}

/**
 * The status of an async video generation task
 */
export enum VideoTaskStatus {
  /** Task is waiting to be processed */
  Pending = 'Pending',
  
  /** Task is currently being processed */
  Running = 'Running',
  
  /** Task completed successfully */
  Completed = 'Completed',
  
  /** Task failed with an error */
  Failed = 'Failed',
  
  /** Task was cancelled */
  Cancelled = 'Cancelled',
  
  /** Task timed out */
  TimedOut = 'TimedOut'
}

/**
 * Options for polling video task status
 */
export interface VideoTaskPollingOptions {
  /** The polling interval in milliseconds */
  intervalMs?: number;
  
  /** The maximum polling timeout in milliseconds */
  timeoutMs?: number;
  
  /** Whether to use exponential backoff for polling intervals */
  useExponentialBackoff?: boolean;
  
  /** The maximum interval between polls in milliseconds when using exponential backoff */
  maxIntervalMs?: number;
}

/**
 * Common video models supported by Conduit
 */
export const VideoModels = {
  /** MiniMax video generation model */
  MINIMAX_VIDEO: 'minimax-video',
  
  /** Default video model */
  DEFAULT: 'minimax-video'
} as const;

/**
 * Common video resolutions
 */
export const VideoResolutions = {
  /** 720p resolution (1280x720) */
  HD: '1280x720',
  
  /** 1080p resolution (1920x1080) */
  FULL_HD: '1920x1080',
  
  /** Vertical 720p (720x1280) */
  VERTICAL_HD: '720x1280',
  
  /** Vertical 1080p (1080x1920) */
  VERTICAL_FULL_HD: '1080x1920',
  
  /** Square format (720x720) */
  SQUARE: '720x720'
} as const;

/**
 * Video response formats
 */
export const VideoResponseFormats = {
  /** Return video as URL (default) */
  URL: 'url',
  
  /** Return video as base64-encoded JSON */
  BASE64_JSON: 'b64_json'
} as const;

/**
 * Default values for video generation
 */
export const VideoDefaults = {
  /** Default duration in seconds */
  DURATION: 5,
  
  /** Default resolution */
  RESOLUTION: VideoResolutions.HD,
  
  /** Default frames per second */
  FPS: 30,
  
  /** Default response format */
  RESPONSE_FORMAT: VideoResponseFormats.URL,
  
  /** Default polling interval in milliseconds */
  POLLING_INTERVAL_MS: 2000,
  
  /** Default polling timeout in milliseconds */
  POLLING_TIMEOUT_MS: 600000, // 10 minutes
  
  /** Default maximum polling interval in milliseconds */
  MAX_POLLING_INTERVAL_MS: 30000 // 30 seconds
} as const;

/**
 * Capabilities of a video generation model
 */
export interface VideoModelCapabilities {
  /** Maximum duration in seconds */
  maxDuration: number;
  
  /** Supported resolutions */
  supportedResolutions: string[];
  
  /** Supported FPS values */
  supportedFps: number[];
  
  /** Whether the model supports custom styles */
  supportsCustomStyles: boolean;
  
  /** Whether the model supports seed-based generation */
  supportsSeed: boolean;
  
  /** Maximum number of videos that can be generated in one request */
  maxVideos: number;
}

/**
 * Gets the capabilities for a specific video model
 */
export function getVideoModelCapabilities(model: string): VideoModelCapabilities {
  const modelLower = model.toLowerCase();
  
  switch (modelLower) {
    case 'minimax-video':
    case 'minimax-video-01':
      return {
        maxDuration: 6,
        supportedResolutions: [
          VideoResolutions.HD,
          VideoResolutions.FULL_HD,
          VideoResolutions.VERTICAL_HD,
          VideoResolutions.VERTICAL_FULL_HD,
          '720x480'
        ],
        supportedFps: [24, 30],
        supportsCustomStyles: true,
        supportsSeed: true,
        maxVideos: 1
      };
    
    default:
      return {
        maxDuration: 60,
        supportedResolutions: [
          VideoResolutions.HD,
          VideoResolutions.FULL_HD,
          VideoResolutions.SQUARE
        ],
        supportedFps: [24, 30, 60],
        supportsCustomStyles: true,
        supportsSeed: true,
        maxVideos: 10
      };
  }
}

/**
 * Validates a video generation request
 */
export function validateVideoGenerationRequest(request: VideoGenerationRequest): void {
  if (!request.prompt || request.prompt.trim().length === 0) {
    throw new Error('Prompt is required');
  }
  
  if (request.n !== undefined && (request.n <= 0 || request.n > 10)) {
    throw new Error('Number of videos must be between 1 and 10');
  }
  
  if (request.duration !== undefined && (request.duration <= 0 || request.duration > 60)) {
    throw new Error('Duration must be between 1 and 60 seconds');
  }
  
  if (request.fps !== undefined && (request.fps <= 0 || request.fps > 120)) {
    throw new Error('FPS must be between 1 and 120');
  }
  
  if (request.response_format && 
      request.response_format !== VideoResponseFormats.URL && 
      request.response_format !== VideoResponseFormats.BASE64_JSON) {
    throw new Error(`Response format must be '${VideoResponseFormats.URL}' or '${VideoResponseFormats.BASE64_JSON}'`);
  }
}

/**
 * Validates an async video generation request
 */
export function validateAsyncVideoGenerationRequest(request: AsyncVideoGenerationRequest): void {
  // First validate the base video generation request
  validateVideoGenerationRequest(request);
  
  // Additional validation for async-specific fields
  if (request.timeout_seconds !== undefined && 
      (request.timeout_seconds <= 0 || request.timeout_seconds > 3600)) {
    throw new Error('Timeout must be between 1 and 3600 seconds');
  }
  
  if (request.webhook_url && !isValidUrl(request.webhook_url)) {
    throw new Error('WebhookUrl must be a valid HTTP or HTTPS URL');
  }
}

/**
 * Helper function to validate URLs
 */
function isValidUrl(url: string): boolean {
  try {
    const parsedUrl = new URL(url);
    return parsedUrl.protocol === 'http:' || parsedUrl.protocol === 'https:';
  } catch {
    return false;
  }
}

File: archive/Clients.DotNet/ConduitLLM.CoreClient/src/Models/Videos.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ConduitLLM.CoreClient.Models
{
    /// <summary>
    /// Represents a request to generate a video based on a text prompt.
    /// </summary>
    public class VideoGenerationRequest
    {
        /// <summary>
        /// Gets or sets the text prompt that describes what video to generate.
        /// </summary>
        [Required]
        [JsonPropertyName("prompt")]
        public string Prompt { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the model to use for video generation (e.g., "minimax-video-01").
        /// </summary>
        [JsonPropertyName("model")]
        public string? Model { get; set; }

        /// <summary>
        /// Gets or sets the duration of the video in seconds. Defaults to 5 seconds.
        /// </summary>
        [Range(1, 60)]
        [JsonPropertyName("duration")]
        public int? Duration { get; set; }

        /// <summary>
        /// Gets or sets the size/resolution of the video (e.g., "1920x1080", "1280x720").
        /// </summary>
        [JsonPropertyName("size")]
        public string? Size { get; set; }

        /// <summary>
        /// Gets or sets frames per second for the video. Common values: 24, 30, 60.
        /// </summary>
        [Range(1, 120)]
        [JsonPropertyName("fps")]
        public int? Fps { get; set; }

        /// <summary>
        /// Gets or sets the style or aesthetic of the video generation.
        /// </summary>
        [JsonPropertyName("style")]
        public string? Style { get; set; }

        /// <summary>
        /// Gets or sets the format in which the generated video is returned.
        /// Options: "url" (default) or "b64_json" (base64 encoded).
        /// </summary>
        [JsonPropertyName("response_format")]
        public string? ResponseFormat { get; set; }

        /// <summary>
        /// Gets or sets a unique identifier representing your end-user.
        /// </summary>
        [JsonPropertyName("user")]
        public string? User { get; set; }

        /// <summary>
        /// Gets or sets optional seed for deterministic generation.
        /// </summary>
        [JsonPropertyName("seed")]
        public int? Seed { get; set; }

        /// <summary>
        /// Gets or sets the number of videos to generate. Defaults to 1.
        /// </summary>
        [Range(1, 10)]
        [JsonPropertyName("n")]
        public int N { get; set; } = 1;
    }

    /// <summary>
    /// Represents the response from a video generation request.
    /// </summary>
    public class VideoGenerationResponse
    {
        /// <summary>
        /// Gets or sets the Unix timestamp of when the response was created.
        /// </summary>
        [JsonPropertyName("created")]
        public long Created { get; set; }

        /// <summary>
        /// Gets or sets the list of generated video data.
        /// </summary>
        [JsonPropertyName("data")]
        public List<VideoData> Data { get; set; } = new();

        /// <summary>
        /// Gets or sets the model used for generation.
        /// </summary>
        [JsonPropertyName("model")]
        public string? Model { get; set; }

        /// <summary>
        /// Gets or sets usage information if available.
        /// </summary>
        [JsonPropertyName("usage")]
        public VideoUsage? Usage { get; set; }
    }

    /// <summary>
    /// Represents a single generated video.
    /// </summary>
    public class VideoData
    {
        /// <summary>
        /// Gets or sets the URL of the generated video, if response_format is "url".
        /// </summary>
        [JsonPropertyName("url")]
        public string? Url { get; set; }

        /// <summary>
        /// Gets or sets the base64-encoded video data, if response_format is "b64_json".
        /// </summary>
        [JsonPropertyName("b64_json")]
        public string? B64Json { get; set; }

        /// <summary>
        /// Gets or sets the revised prompt that was used for generation.
        /// </summary>
        [JsonPropertyName("revised_prompt")]
        public string? RevisedPrompt { get; set; }

        /// <summary>
        /// Gets or sets additional metadata about the generated video.
        /// </summary>
        [JsonPropertyName("metadata")]
        public VideoMetadata? Metadata { get; set; }
    }

    /// <summary>
    /// Represents usage statistics for video generation.
    /// </summary>
    public class VideoUsage
    {
        /// <summary>
        /// Gets or sets the number of prompt tokens used.
        /// </summary>
        [JsonPropertyName("prompt_tokens")]
        public int PromptTokens { get; set; }

        /// <summary>
        /// Gets or sets the total number of tokens used.
        /// </summary>
        [JsonPropertyName("total_tokens")]
        public int TotalTokens { get; set; }

        /// <summary>
        /// Gets or sets the duration processed in seconds.
        /// </summary>
        [JsonPropertyName("duration_seconds")]
        public double? DurationSeconds { get; set; }

        /// <summary>
        /// Gets or sets the total processing time in seconds.
        /// </summary>
        [JsonPropertyName("processing_time_seconds")]
        public double? ProcessingTimeSeconds { get; set; }
    }

    /// <summary>
    /// Represents metadata about a generated video.
    /// </summary>
    public class VideoMetadata
    {
        /// <summary>
        /// Gets or sets the actual duration of the generated video in seconds.
        /// </summary>
        [JsonPropertyName("duration")]
        public double? Duration { get; set; }

        /// <summary>
        /// Gets or sets the resolution of the generated video.
        /// </summary>
        [JsonPropertyName("resolution")]
        public string? Resolution { get; set; }

        /// <summary>
        /// Gets or sets the frames per second of the generated video.
        /// </summary>
        [JsonPropertyName("fps")]
        public int? Fps { get; set; }

        /// <summary>
        /// Gets or sets the file size in bytes.
        /// </summary>
        [JsonPropertyName("file_size_bytes")]
        public long? FileSizeBytes { get; set; }

        /// <summary>
        /// Gets or sets the video format/codec.
        /// </summary>
        [JsonPropertyName("format")]
        public string? Format { get; set; }

        /// <summary>
        /// Gets or sets the video codec used for encoding.
        /// </summary>
        [JsonPropertyName("codec")]
        public string? Codec { get; set; }

        /// <summary>
        /// Gets or sets the audio codec used for encoding.
        /// </summary>
        [JsonPropertyName("audio_codec")]
        public string? AudioCodec { get; set; }

        /// <summary>
        /// Gets or sets the bitrate of the video.
        /// </summary>
        [JsonPropertyName("bitrate")]
        public int? Bitrate { get; set; }

        /// <summary>
        /// Gets or sets the MIME type of the video file.
        /// </summary>
        [JsonPropertyName("mime_type")]
        public string? MimeType { get; set; }

        /// <summary>
        /// Gets or sets the seed used for generation, if any.
        /// </summary>
        [JsonPropertyName("seed")]
        public int? Seed { get; set; }
    }

    /// <summary>
    /// Represents an async video generation request.
    /// </summary>
    public class AsyncVideoGenerationRequest : VideoGenerationRequest
    {
        /// <summary>
        /// Gets or sets the webhook URL to receive the result when generation is complete.
        /// </summary>
        [JsonPropertyName("webhook_url")]
        public string? WebhookUrl { get; set; }

        /// <summary>
        /// Gets or sets additional metadata to include with the webhook callback.
        /// </summary>
        [JsonPropertyName("webhook_metadata")]
        public Dictionary<string, object>? WebhookMetadata { get; set; }

        /// <summary>
        /// Gets or sets additional headers to include with the webhook callback.
        /// </summary>
        [JsonPropertyName("webhook_headers")]
        public Dictionary<string, string>? WebhookHeaders { get; set; }

        /// <summary>
        /// Gets or sets the timeout for the generation task in seconds.
        /// </summary>
        [Range(1, 3600)]
        [JsonPropertyName("timeout_seconds")]
        public int? TimeoutSeconds { get; set; }
    }

    /// <summary>
    /// Represents the response from an async video generation request.
    /// </summary>
    public class AsyncVideoGenerationResponse
    {
        /// <summary>
        /// Gets or sets the unique task identifier.
        /// </summary>
        [JsonPropertyName("task_id")]
        public string TaskId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the current status of the task.
        /// </summary>
        [JsonPropertyName("status")]
        public VideoTaskStatus Status { get; set; }

        /// <summary>
        /// Gets or sets the progress percentage (0-100).
        /// </summary>
        [JsonPropertyName("progress")]
        public int Progress { get; set; }

        /// <summary>
        /// Gets or sets an optional progress message.
        /// </summary>
        [JsonPropertyName("message")]
        public string? Message { get; set; }

        /// <summary>
        /// Gets or sets the estimated time to completion in seconds.
        /// </summary>
        [JsonPropertyName("estimated_time_to_completion")]
        public int? EstimatedTimeToCompletion { get; set; }

        /// <summary>
        /// Gets or sets when the task was created.
        /// </summary>
        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Gets or sets when the task was last updated.
        /// </summary>
        [JsonPropertyName("updated_at")]
        public DateTime UpdatedAt { get; set; }

        /// <summary>
        /// Gets or sets the generation result, available when status is Completed.
        /// </summary>
        [JsonPropertyName("result")]
        public VideoGenerationResponse? Result { get; set; }

        /// <summary>
        /// Gets or sets error information if the task failed.
        /// </summary>
        [JsonPropertyName("error")]
        public string? Error { get; set; }
    }

    /// <summary>
    /// Represents the status of an async video generation task.
    /// </summary>
    public enum VideoTaskStatus
    {
        /// <summary>
        /// Task is waiting to be processed.
        /// </summary>
        Pending,

        /// <summary>
        /// Task is currently being processed.
        /// </summary>
        Running,

        /// <summary>
        /// Task completed successfully.
        /// </summary>
        Completed,

        /// <summary>
        /// Task failed with an error.
        /// </summary>
        Failed,

        /// <summary>
        /// Task was cancelled.
        /// </summary>
        Cancelled,

        /// <summary>
        /// Task timed out.
        /// </summary>
        TimedOut
    }

    /// <summary>
    /// Represents options for polling video task status.
    /// </summary>
    public class VideoTaskPollingOptions
    {
        /// <summary>
        /// Gets or sets the polling interval in milliseconds.
        /// </summary>
        public int IntervalMs { get; set; } = 2000;

        /// <summary>
        /// Gets or sets the maximum polling timeout in milliseconds.
        /// </summary>
        public int TimeoutMs { get; set; } = 600000; // 10 minutes

        /// <summary>
        /// Gets or sets whether to use exponential backoff for polling intervals.
        /// </summary>
        public bool UseExponentialBackoff { get; set; } = true;

        /// <summary>
        /// Gets or sets the maximum interval between polls in milliseconds when using exponential backoff.
        /// </summary>
        public int MaxIntervalMs { get; set; } = 30000; // 30 seconds
    }

    /// <summary>
    /// Common video models supported by Conduit.
    /// </summary>
    public static class VideoModels
    {
        /// <summary>
        /// MiniMax video generation model.
        /// </summary>
        public const string MiniMaxVideo = "minimax-video";

        /// <summary>
        /// Default video model.
        /// </summary>
        public const string Default = MiniMaxVideo;
    }

    /// <summary>
    /// Common video resolutions.
    /// </summary>
    public static class VideoResolutions
    {
        /// <summary>
        /// 720p resolution (1280x720).
        /// </summary>
        public const string HD = "1280x720";

        /// <summary>
        /// 1080p resolution (1920x1080).
        /// </summary>
        public const string FullHD = "1920x1080";

        /// <summary>
        /// Vertical 720p (720x1280).
        /// </summary>
        public const string VerticalHD = "720x1280";

        /// <summary>
        /// Vertical 1080p (1080x1920).
        /// </summary>
        public const string VerticalFullHD = "1080x1920";

        /// <summary>
        /// Square format (720x720).
        /// </summary>
        public const string Square = "720x720";
    }

    /// <summary>
    /// Video response formats.
    /// </summary>
    public static class VideoResponseFormats
    {
        /// <summary>
        /// Return video as URL (default).
        /// </summary>
        public const string Url = "url";

        /// <summary>
        /// Return video as base64-encoded JSON.
        /// </summary>
        public const string Base64Json = "b64_json";
    }

    /// <summary>
    /// Default values for video generation.
    /// </summary>
    public static class VideoDefaults
    {
        /// <summary>
        /// Default duration in seconds.
        /// </summary>
        public const int Duration = 5;

        /// <summary>
        /// Default resolution.
        /// </summary>
        public const string Resolution = VideoResolutions.HD;

        /// <summary>
        /// Default frames per second.
        /// </summary>
        public const int Fps = 30;

        /// <summary>
        /// Default response format.
        /// </summary>
        public const string ResponseFormat = VideoResponseFormats.Url;
    }

    /// <summary>
    /// Base class for webhook payloads sent by Conduit.
    /// </summary>
    public abstract class WebhookPayloadBase
    {
        /// <summary>
        /// Gets or sets the unique identifier for this webhook event.
        /// </summary>
        [JsonPropertyName("event_id")]
        public string EventId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the type of webhook event.
        /// </summary>
        [JsonPropertyName("event_type")]
        public string EventType { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the timestamp when the event occurred.
        /// </summary>
        [JsonPropertyName("timestamp")]
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets or sets the task ID associated with this event.
        /// </summary>
        [JsonPropertyName("task_id")]
        public string TaskId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets optional metadata provided in the original request.
        /// </summary>
        [JsonPropertyName("metadata")]
        public Dictionary<string, object>? Metadata { get; set; }
    }

    /// <summary>
    /// Webhook payload sent when video generation is completed.
    /// </summary>
    public class VideoCompletionWebhookPayload : WebhookPayloadBase
    {
        /// <summary>
        /// Gets or sets the final status of the video generation task.
        /// </summary>
        [JsonPropertyName("status")]
        public VideoTaskStatus Status { get; set; }

        /// <summary>
        /// Gets or sets the generated video result, if successful.
        /// </summary>
        [JsonPropertyName("result")]
        public VideoGenerationResponse? Result { get; set; }

        /// <summary>
        /// Gets or sets error information if the task failed.
        /// </summary>
        [JsonPropertyName("error")]
        public string? Error { get; set; }

        /// <summary>
        /// Gets or sets the total processing time in seconds.
        /// </summary>
        [JsonPropertyName("processing_time_seconds")]
        public double? ProcessingTimeSeconds { get; set; }
    }

    /// <summary>
    /// Webhook payload sent to provide progress updates during video generation.
    /// </summary>
    public class VideoProgressWebhookPayload : WebhookPayloadBase
    {
        /// <summary>
        /// Gets or sets the current status of the video generation task.
        /// </summary>
        [JsonPropertyName("status")]
        public VideoTaskStatus Status { get; set; }

        /// <summary>
        /// Gets or sets the progress percentage (0-100).
        /// </summary>
        [JsonPropertyName("progress")]
        public int Progress { get; set; }

        /// <summary>
        /// Gets or sets an optional progress message.
        /// </summary>
        [JsonPropertyName("message")]
        public string? Message { get; set; }

        /// <summary>
        /// Gets or sets the estimated time to completion in seconds.
        /// </summary>
        [JsonPropertyName("estimated_time_to_completion")]
        public int? EstimatedTimeToCompletion { get; set; }
    }
}